velox/functions/prestosql/window/Rank.cpp

```diff
@@ -75,10 +113,21 @@ void registerRank(const std::string& name) {
       [name](
           const std::vector<TypePtr>& /*argTypes*/,
           const std::vector<column_index_t>& /*argIndices*/,
-          const TypePtr& /*resultType*/,
+          const TypePtr& resultType,
           velox::memory::MemoryPool* /*pool*/)
           -> std::unique_ptr<exec::WindowFunction> {
-        return std::make_unique<RankFunction>();
+        return std::make_unique<RankFunction<R>>(resultType);
       });
 }
+
+template void registerRank<velox::exec::RankType::kRank>(
```

Comment:
There is one more change we can make here to make this more cleaner.

The registerRank changes you have made here are fine. Maybe change the name to registerRankInternal.

Then have a function called
registerRank(...) {
  registerRankInternal<...kRank>(...)
}

registerPercentRank(...) {
  registerRankInternal<...kPercentRank>(...)
}

The registerRank, registerPercentRank, etc are visible in WindowsRegistration and can be used there.

We shouldn't expose the RankType enum outside this file.

Reply:
Thanks for the suggestion!